definitions/npm/react-i18next_v1.10.x/flow_v0.36.x-/react-i18next_v1.10.x.js

```diff
@@ -0,0 +1,35 @@
+import type { TFunction } from 'i18next'
+
+declare module 'react-i18next' {
+  declare type TFunction = TFunction;
+  declare type Locales = string | Array<string>;
+
+  declare type StatelessComponent<P> = (props: P) => ?React$Element<any>;
+
+  declare type Comp<P> = StatelessComponent<P> | Class<React$Component<*, P, *>>;
+
+  declare type Translator<OP, P> = {
+    (component: StatelessComponent<P>): Class<React$Component<void, OP, void>>;
+    <Def, St>(component: Class<React$Component<Def, P, St>>): Class<React$Component<Def, OP, St>>;
```

Comment:
Are you confused by the `<Def, St>` syntax? These are generic placeholders... just imagine those as type variables, which will be substituted in specific calls... you don't really care about the concrete types, you just want them to be the same type consistently for this function call....

e.g.
```

// Generic N -> we don't care about the type,
// we just want the `test` parameter and the return type
// to be the same type
function foo<N>(test: N): N {
   return test;
}

// test => Number... return type => Number... t => Number... works
const t: number = foo(1);

// N => Number... return type => Number.... t => String ... doesn't work
const t: string = foo(1);
```

does this help?